dependencies/boon/CMakeLists.txt

```diff
@@ -0,0 +1,28 @@
+find_package(Cargo)
+if(NOT Cargo_FOUND)
```

Comment:
I am a bit concerned about making this cargo dependency mandatory for something that is only used at build time. Users that don't have rust installed should still be able to build the model and run it. Can we make this either 1) completely optional or 2) require a -DSCHEMA_VALIDATION=OFF flag?

Reply:
#1189 will require boon to be installed for run-time validation.  I'd prefer run-time schema validation to not be optional.

Reply:
I made this completely optional.

Using a library approach would require the emulator to be built and then executed to validate the generated configurations (including the default config). In case that turns out to be overly complicated, I updated this PR and made the boon dependency optional, and enabled it in CI.